### PR TITLE
added directories and made AMG test parallel

### PR DESCRIPTION
**Pull request by ckendric**
The amg files are included here. The yaml file runs both of AMG's built-in problems in a parallel way. It records the number of iterations and the Figure of Merit for each problem.

### CODE REVIEW

File: ProxyAppsSuite/tests/AMG.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    inherits_from: _base
+    summary: running AMG's built in problem1
+    variables:
+        nvals: { nxval: '100', nyval: '100', nzval: '100' }
+        pvals: { pxval: '2', pyval: '2', pzval: '1' }
+    run:
+        cmds:
+            - cd test
+            - echo "running problem one"
+            - srun -n 4 amg -P "{{pvals.pxval}}" "{{pvals.pyval}}" "{{pvals.pzval}}" -n "{{nvals.nxval}}" "{{nvals.nyval}}" "{{nvals.nzval}}" &> problem1.out
```
Comment (francinelapid) [inline ProxyAppsSuite/tests/AMG.yaml:28]: I don't think you need the `-n 4` here, as you already specified `tasks_per_node: 4`

File: ProxyAppsSuite/tests/AMG.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        source_path: amg.tar.gz
+        env:
+            CC: mpicc
+        cmds:
+            - make all
+
+    scheduler: slurm
+    slurm:
+        num_nodes: 1
+        tasks_per_node: 4
```
Comment (francinelapid) [inline ProxyAppsSuite/tests/AMG.yaml:16]: Is this a test we can do using multiple nodes or perhaps run once per node?

**Review by francinelapid — COMMENTED** [review]
Looks good so far! Could y'all make the summaries more descriptive, perhaps maybe specify what kind of problem it's running?

File: ProxyAppsSuite/tests/AMG.yaml (hunk 1 above)
Comment (jennfshr) [reply ProxyAppsSuite/tests/AMG.yaml:28]: @francinelapid I just confirmed that you're correct! shouldn't need to specify `-n 4` on the run cmd line as long as there are tasks_per_node specified.

**Pull request merged**